webpack/scenes/ContentViews/Details/Versions/Delete/RemoveSteps/CVEnvironmentSelectionForm.js

```diff
@@ -0,0 +1,125 @@
+import React, { useEffect, useState, useContext } from 'react';
+import { Alert, Checkbox, EmptyState, EmptyStateVariant, Title, EmptyStateBody } from '@patternfly/react-core';
+import { TableVariant, TableComposable, Thead, Tbody, Tr, Th, Td } from '@patternfly/react-table';
+import { translate as __ } from 'foremanReact/common/I18n';
+import DeleteContext from '../DeleteContext';
+
+const CVEnvironmentSelectionForm = () => {
+  const {
+    versionNameToRemove, versionEnvironments, selected, setSelected,
+    setAffectedActivationKeys, setAffectedHosts, deleteFlow, setDeleteFlow,
+  } = useContext(DeleteContext);
+
+  const [allRowsSelected, setAllRowsSelected] = useState(false);
+
+  const onSelect = (event, isSelected, rowId) => {
+    const newSelected = selected.map((sel, index) => (index === rowId ? isSelected : sel));
+    setSelected(newSelected);
+
+    if (!isSelected && allRowsSelected) {
+      setAllRowsSelected(false);
+      setDeleteFlow(false);
+    } else if (isSelected && !allRowsSelected) {
+      let allSelected = true;
+      for (let i = 0; i < selected.length; i += 1) {
+        if (i !== rowId) {
+          if (!selected[i]) {
+            allSelected = false;
+          }
+        }
+      }
+      if (allSelected) {
+        setAllRowsSelected(true);
+      }
+    }
```

Comment:
Instead of a state value that you have to keep up to date, I feel like `allRowsSelected` could be a simpler function (lazily evaluated) that you don't have to update every time like this.  (especially if you use `useSet` :)